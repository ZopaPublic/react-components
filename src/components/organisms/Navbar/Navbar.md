Basic example:

```js { "props": { "style": { "transform": "translate3d(0, 0, 0)" } } }
import { Navbar, Link, HamburgerIcon, ProfileIcon } from '@zopauk/react-components';

<Navbar.Layout
  backgroundColor="#00B9A7"
  left={
    <Navbar.Dropdown
      id="unique-id"
      ariaLabel="Awesome navbar dropdown"
      items={[{ label: 'one', href: '#' }, { label: 'two', href: '#' }, { label: 'three', href: '#' }]}
      renderOpener={({ open, getOpenerProps }) => (
        <div style={{ padding: 8 }}>
          <Navbar.Link href="#" open={open} {...getOpenerProps()}>
            <HamburgerIcon size="30px" activeColor="#fff" inactiveColor="#fff" />
          </Navbar.Link>
        </div>
      )}
      renderItem={({ item: { label, href }, getItemProps }) => (
        <Link href={href} {...getItemProps()}>
          {label}
        </Link>
      )}
    />
  }
  right={
    <Navbar.Dropdown
      id="another-unique-id"
      ariaLabel="Awesome navbar dropdown"
      items={[{ label: 'one', href: '#' }, { label: 'two', href: '#' }]}
      renderOpener={({ open, getOpenerProps }) => {
        return (
          <div style={{ padding: 8 }}>
            <Navbar.Link color="#fff" href="#" open={open} {...getOpenerProps()}>
              <ProfileIcon activeColor="#fff" inactiveColor="#fff" />
            </Navbar.Link>
          </div>
        );
      }}
      renderItem={({ item: { label, href }, getItemProps }) => (
        <Link href={href} {...getItemProps()}>
          {label}
        </Link>
      )}
    />
  }
/>;
```

Responsive Navbar example:

```js { "props": { "style": { "transform": "translate3d(0, 0, 0)" } } }
import { Navbar, Link, HamburgerIcon, FlexCol } from '@zopauk/react-components';

class ResponsiveNavbar extends React.Component {
  constructor(props) {
    super(props);
    this.renderLeft = this.renderLeft.bind(this);
  }

  renderLeft() {
    const items = [{ label: 'one', href: '#' }, { label: 'two', href: '#' }, { label: 'three', href: '#' }];
    return (
      <div>
        <FlexCol xs="auto" m="hidden">
          <Navbar.Dropdown
            id="yet-another-unique-id"
            ariaLabel="Awesome navbar dropdown"
            items={items}
            renderOpener={({ open, getOpenerProps }) => (
              <div style={{ padding: 8 }}>
                <Navbar.Link color="#fff" href="#" open={open} {...getOpenerProps()}>
                  <HamburgerIcon size="30px" activeColor="#fff" inactiveColor="#fff" />
                </Navbar.Link>
              </div>
            )}
            renderItem={({ item: { label, href }, getItemProps }) => (
              <Link href={href} {...getItemProps()}>
                {label}
              </Link>
            )}
          />
        </FlexCol>
        <FlexCol xs="hidden" m="auto">
          {items.map(({ label, href }, index) => (
            <Link key={`desktop-link-${index}`} href={href} style={{ padding: 8 }}>
              {label}
            </Link>
          ))}
        </FlexCol>
      </div>
    );
  }

  render() {
    return <Navbar.Layout backgroundColor="#00B9A7" left={this.renderLeft()} />;
  }
}

<ResponsiveNavbar />;
```
