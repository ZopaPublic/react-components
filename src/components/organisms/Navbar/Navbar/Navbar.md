### Summary

Navbar will render an array of links provided using a dropdown which works on click and a link which is a wrap around our [`<Link />`](/#/Components/Atoms/Link) atom, decorated with tow extra props: `active` and `withChevron`. You can opt out displaying the CTA using `withCTA` prop.

⚠️ The menu animates when scrolling. For a better preview view [Default example](#/Components/Organisms/Navbar/Navbar/1)

### Examples

- Default theme

```ts { "props": { "style": { "transform": "translate3d(0, 0, 0)", "backgroundColor": "#FFFFFF", "overflow": "hidden", "height": "500px", "padding": "0" } } }
import { Navbar } from '@zopauk/react-components';

const NAV_ITEMS = [
  {
    label: 'Borrow',
    'data-automation': 'Borrow.topBar.Button',
    children: [
      {
        label: 'Loans',
        href: '/loans',
        onClick: () => alert('testing'),
        'data-automation': 'loans.Borrow.topBar.Menu',
      },
      {
        label: 'Car loans',
        href: '/loans/car-loans',
        'data-automation': 'Car_loan.Borrow.topBar.Menu',
      },
      {
        label: 'Borrowing power',
        href: '/borrowing-power',
        'data-automation': 'Borrowing_Power.Borrow.topBar.Menu',
      },
    ],
  },
  {
    label: 'Invest',
    'data-automation': 'Invest.topBar.Menu',
    children: [
      {
        label: 'Peer-to peer investments',
        href: '/invest',
        'data-automation': 'Peer_to_peer.Invest.topBar.Menu',
      },
      {
        label: 'Innovative Finance ISA',
        href: '/invest/isa',
        'data-automation': 'Innovative.Invest.topBar.Menu',
      },
    ],
  },
  {
    label: 'Save',
    href: '/savings-accounts',
    'data-automation': 'Save.topBar.Menu',
  },
  {
    label: 'About',
    'data-automation': 'About.topBar.Menu',
    children: [
      {
        label: 'About Zopa',
        href: '/about',
        'data-automation': 'About.About.topBar.Menu',
      },
      {
        label: 'Our story',
        href: '/about/our-story',
        'data-automation': 'Story.About.topBar.Menu',
      },
      {
        label: 'Our board',
        href: '/about/board',
        'data-automation': 'Our_Board.About.topBar.Menu',
      },
      {
        label: 'Our leadership team',
        href: '/about/leadership',
        'data-automation': 'Our_Leadership.About.topBar.Menu',
      },
      {
        label: 'Awards',
        href: '/about/awards',
        'data-automation': 'Awards.About.topBar.Menu',
      },
      {
        label: 'Careers',
        href: '/about/careers',
        'data-automation': 'Careers.About.topBar.Menu',
      },
      {
        label: 'Press office',
        href: '/about/press',
        'data-automation': 'Press_office.About.topBar.Menu',
      },
    ],
  },
  {
    label: 'Support',
    href: '/contact',
    'data-automation': 'Support.topBar.Menu',
  },
];

<>
  <Navbar overlayLogoWith={<span></span>} links={NAV_ITEMS} />
  <div>
    <h1>This is a heading</h1>
    <p>This is a paragraph</p>
  </div>
</>;
```

- Collapsed theme

```ts { "props": { "style": { "transform": "translate3d(0, 0, 0)", "backgroundColor": "#FFFFFF", "overflow": "hidden", "height": "500px", "padding": "0" } } }
import { Navbar } from '@zopauk/react-components';

const NAV_ITEMS = [
  {
    label: 'Borrow',
    'data-automation': 'Borrow.topBar.Button',
    children: [
      {
        label: 'Loans',
        href: '/loans',
        onClick: () => alert('testing'),
        'data-automation': 'loans.Borrow.topBar.Menu',
      },
      {
        label: 'Car loans',
        href: '/loans/car-loans',
        'data-automation': 'Car_loan.Borrow.topBar.Menu',
      },
      {
        label: 'Borrowing power',
        href: '/borrowing-power',
        'data-automation': 'Borrowing_Power.Borrow.topBar.Menu',
      },
    ],
  },
  {
    label: 'Invest',
    'data-automation': 'Invest.topBar.Menu',
    children: [
      {
        label: 'Peer-to peer investments',
        href: '/invest',
        'data-automation': 'Peer_to_peer.Invest.topBar.Menu',
      },
      {
        label: 'Innovative Finance ISA',
        href: '/invest/isa',
        'data-automation': 'Innovative.Invest.topBar.Menu',
      },
    ],
  },
  {
    label: 'About',
    'data-automation': 'About.topBar.Menu',
    children: [
      {
        label: 'About Zopa',
        href: '/about',
        'data-automation': 'About.About.topBar.Menu',
      },
      {
        label: 'Our story',
        href: '/about/our-story',
        'data-automation': 'Story.About.topBar.Menu',
      },
      {
        label: 'Our board',
        href: '/about/board',
        'data-automation': 'Our_Board.About.topBar.Menu',
      },
      {
        label: 'Our leadership team',
        href: '/about/leadership',
        'data-automation': 'Our_Leadership.About.topBar.Menu',
      },
      {
        label: 'Awards',
        href: '/about/awards',
        'data-automation': 'Awards.About.topBar.Menu',
      },
      {
        label: 'Careers',
        href: '/about/careers',
        'data-automation': 'Careers.About.topBar.Menu',
      },
      {
        label: 'Press office',
        href: '/about/press',
        'data-automation': 'Press_office.About.topBar.Menu',
      },
    ],
  },
  {
    label: 'Support',
    href: '/contact',
    'data-automation': 'Support.topBar.Menu',
  },
];

<>
  <Navbar overlayLogoWith={<span></span>} links={NAV_ITEMS} collapsed />
  <div>
    <h1>This is a heading</h1>
    <p>This is a paragraph</p>
  </div>
</>;
```

- With sub headings

```ts { "props": { "style": { "transform": "translate3d(0, 0, 0)", "backgroundColor": "#FFFFFF", "overflow": "hidden", "height": "500px", "padding": "0" } } }
import { Navbar } from '@zopauk/react-components';

const NAV_ITEMS = [
  {
    label: 'Borrow',
    'data-automation': 'Borrow.topBar.Button',
    children: [
      {
        label: 'Loans',
        href: '/loans',
        'data-automation': 'loans.Borrow.topBar.Menu',
      },
      {
        label: 'Car finance',
        isDropdownHeading: true,
      },
      {
        label: 'Car loans',
        href: '/loans/car-loans',
        'data-automation': 'Car_loan.Borrow.topBar.Menu',
      },
      {
        label: 'Tools',
        isDropdownHeading: true,
      },
      {
        label: 'Borrowing power',
        href: '/borrowing-power',
        'data-automation': 'Borrowing_Power.Borrow.topBar.Menu',
      },
    ],
  },
  {
    label: 'Invest',
    'data-automation': 'Invest.topBar.Menu',
    children: [
      {
        label: 'Peer-to peer investments',
        href: '/invest',
        'data-automation': 'Peer_to_peer.Invest.topBar.Menu',
      },
      {
        label: 'Innovative Finance ISA',
        href: '/invest/isa',
        'data-automation': 'Innovative.Invest.topBar.Menu',
      },
    ],
  },
  {
    label: 'About',
    'data-automation': 'About.topBar.Menu',
    children: [
      {
        label: 'About Zopa',
        href: '/about',
        'data-automation': 'About.About.topBar.Menu',
      },
      {
        label: 'Our story',
        href: '/about/our-story',
        'data-automation': 'Story.About.topBar.Menu',
      },
      {
        label: 'Our board',
        href: '/about/board',
        'data-automation': 'Our_Board.About.topBar.Menu',
      },
      {
        label: 'Our leadership team',
        href: '/about/leadership',
        'data-automation': 'Our_Leadership.About.topBar.Menu',
      },
      {
        label: 'Awards',
        href: '/about/awards',
        'data-automation': 'Awards.About.topBar.Menu',
      },
      {
        label: 'Careers',
        href: '/about/careers',
        'data-automation': 'Careers.About.topBar.Menu',
      },
      {
        label: 'Press office',
        href: '/about/press',
        'data-automation': 'Press_office.About.topBar.Menu',
      },
    ],
  },
  {
    label: 'Support',
    href: '/contact',
    'data-automation': 'Support.topBar.Menu',
  },
];

<>
  <Navbar overlayLogoWith={<span></span>} links={NAV_ITEMS} />
  <div>
    <h1>This is a heading</h1>
    <p>This is a paragraph</p>
  </div>
</>;
```

- Custom Navbar components

```ts { "props": { "style": { "transform": "translate3d(0, 0, 0)", "backgroundColor": "#FFFFFF", "overflow": "hidden", "height": "500px", "padding": "0" } } }
import { forwardRef } from 'react';
import styled from 'styled-components';

import { Navbar, Link, navbarLinkStyles } from '@zopauk/react-components';

const NAV_ITEMS = [
  {
    label: 'Borrow',
    'data-automation': 'Borrow.topBar.Button',
    children: [
      {
        label: 'Loans',
        href: '/loans',
        onClick: () => alert('testing'),
        'data-automation': 'loans.Borrow.topBar.Menu',
      },
      {
        label: 'Car Finance',
        isDropdownHeading: true,
      },
      {
        label: 'Car loans',
        href: '/loans/car-loans',
        'data-automation': 'Car_loan.Borrow.topBar.Menu',
      },
      {
        label: 'Borrowing power',
        href: '/borrowing-power',
        'data-automation': 'Borrowing_Power.Borrow.topBar.Menu',
      },
    ],
  },
  {
    label: 'Invest',
    'data-automation': 'Invest.topBar.Menu',
    children: [
      {
        label: 'Peer-to peer investments',
        href: '/invest',
        'data-automation': 'Peer_to_peer.Invest.topBar.Menu',
      },
      {
        label: 'Innovative Finance ISA',
        href: '/invest/isa',
        'data-automation': 'Innovative.Invest.topBar.Menu',
      },
    ],
  },
  {
    label: 'About',
    'data-automation': 'About.topBar.Menu',
    children: [
      {
        label: 'About Zopa',
        href: '/about',
        'data-automation': 'About.About.topBar.Menu',
      },
      {
        label: 'Our story',
        href: '/about/our-story',
        'data-automation': 'Story.About.topBar.Menu',
      },
      {
        label: 'Our board',
        href: '/about/board',
        'data-automation': 'Our_Board.About.topBar.Menu',
      },
      {
        label: 'Our leadership team',
        href: '/about/leadership',
        'data-automation': 'Our_Leadership.About.topBar.Menu',
      },
      {
        label: 'Awards',
        href: '/about/awards',
        'data-automation': 'Awards.About.topBar.Menu',
      },
      {
        label: 'Careers',
        href: '/about/careers',
        'data-automation': 'Careers.About.topBar.Menu',
      },
      {
        label: 'Press office',
        href: '/about/press',
        'data-automation': 'Press_office.About.topBar.Menu',
      },
    ],
  },
  {
    label: 'Support',
    href: '/contact',
    'data-automation': 'Support.topBar.Menu',
  },
];

// could be a gatsby or react-router-dom <Link />
const SomeLibraryLink = React.forwardRef((props, ref) => {
  const linkProps = Object.assign({}, props);
  delete linkProps.isDropdownLink;

  return <a {...linkProps} ref={ref} />;
});

const CustomLink = styled(SomeLibraryLink)`
  ${navbarLinkStyles}
`;

<Navbar
  overlayLogoWith={<a href="https://www.zopa.com" />}
  links={NAV_ITEMS}
  renderLink={(item, index, props) => (
    <CustomLink {...item} target="_blank" {...props} className="px-4 py-2">
      {item.label}
    </CustomLink>
  )}
/>;
```

- Without CTA

```ts { "props": { "style": { "transform": "translate3d(0, 0, 0)", "backgroundColor": "#FFFFFF", "overflow": "hidden", "height": "500px", "padding": "0" } } }
import { Navbar } from '@zopauk/react-components';

const NAV_ITEMS = [
  {
    label: 'Borrow',
    'data-automation': 'Borrow.topBar.Button',
    children: [
      {
        label: 'Loans',
        href: '/loans',
        onClick: () => alert('testing'),
        'data-automation': 'loans.Borrow.topBar.Menu',
      },
      {
        label: 'Car loans',
        href: '/loans/car-loans',
        'data-automation': 'Car_loan.Borrow.topBar.Menu',
      },
      {
        label: 'Borrowing power',
        href: '/borrowing-power',
        'data-automation': 'Borrowing_Power.Borrow.topBar.Menu',
      },
    ],
  },
  {
    label: 'Invest',
    'data-automation': 'Invest.topBar.Menu',
    children: [
      {
        label: 'Peer-to peer investments',
        href: '/invest',
        'data-automation': 'Peer_to_peer.Invest.topBar.Menu',
      },
      {
        label: 'Innovative Finance ISA',
        href: '/invest/isa',
        'data-automation': 'Innovative.Invest.topBar.Menu',
      },
    ],
  },
  {
    label: 'About',
    'data-automation': 'About.topBar.Menu',
    children: [
      {
        label: 'About Zopa',
        href: '/about',
        'data-automation': 'About.About.topBar.Menu',
      },
      {
        label: 'Our story',
        href: '/about/our-story',
        'data-automation': 'Story.About.topBar.Menu',
      },
      {
        label: 'Our board',
        href: '/about/board',
        'data-automation': 'Our_Board.About.topBar.Menu',
      },
      {
        label: 'Our leadership team',
        href: '/about/leadership',
        'data-automation': 'Our_Leadership.About.topBar.Menu',
      },
      {
        label: 'Awards',
        href: '/about/awards',
        'data-automation': 'Awards.About.topBar.Menu',
      },
      {
        label: 'Careers',
        href: '/about/careers',
        'data-automation': 'Careers.About.topBar.Menu',
      },
      {
        label: 'Press office',
        href: '/about/press',
        'data-automation': 'Press_office.About.topBar.Menu',
      },
    ],
  },
  {
    label: 'Support',
    href: '/contact',
    'data-automation': 'Support.topBar.Menu',
  },
];

<Navbar withCTA={false} links={NAV_ITEMS} />;
```

- Without links

```ts { "props": { "style": { "transform": "translate3d(0, 0, 0)", "backgroundColor": "#FFFFFF", "overflow": "hidden", "height": "500px", "padding": "0" } } }
import { Navbar } from '@zopauk/react-components';

<Navbar overlayLogoWith={<a href="https://www.zopa.com" />}></Navbar>;
```

- Custom CTA

```ts { "props": { "style": { "transform": "translate3d(0, 0, 0)", "backgroundColor": "#FFFFFF", "overflow": "hidden", "height": "500px", "padding": "0" } } }
import { Navbar, Link, buttonStyle } from '@zopauk/react-components';
import styled from 'styled-components';

const ButtonLink = styled(Link)`
  ${buttonStyle}
  margin: 8px;
`;

<Navbar
  overlayLogoWith={<a href="https://www.zopa.com" />}
  cta={
    <ButtonLink href="https://www.zopa.com" styling="primary" data-automation="My_Account.topBar.Menu">
      Sign-in
    </ButtonLink>
  }
></Navbar>;
```
