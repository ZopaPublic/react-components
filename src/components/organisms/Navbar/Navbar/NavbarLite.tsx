import React, { Dispatch, SetStateAction } from 'react';
import styled, { css } from 'styled-components';
import Headroom from 'react-headroom';

import { colors, navbarOpenHeight, navbarClosedHeight, mobileNavbarHeight, spacing } from '../../../../constants';
import { maxMedia, minMedia } from '../../../../helpers/responsiveness';
import useScrollThreshold from '../useScrollThreshold/useScrollThreshold';
import { NavbarLinkProps } from '../NavbarLink/NavbarLink';
import Button from '../../../atoms/Button/Button';
import { AppThemeProps, useThemeContext } from '../../../styles/Theme';
import { ConditionalWrapper } from '../../../atoms/ConditionalWrapper/ConditionalWrapper';

export interface NavigationItem {
  label: React.ReactNode;
  href?: string;
  'data-automation'?: string;
  onClick?: (event?: React.MouseEvent<HTMLAnchorElement>) => void;
  isDropdownHeading?: boolean;
  children?: Exclude<NavigationItem, 'children'>[];
}
export interface NavbarLinksListProps {
  /**
   * Array of links
   */
  links?: NavigationItem[];
  /**
   * Link component
   */
  renderLink?: (item: NavigationItem, index: number, props?: any) => React.ReactNode;
  setOpen: Dispatch<SetStateAction<boolean>>;
}

export interface NavbarProps extends Omit<NavbarLinksListProps, 'setOpen'> {
  /**
   * allows you to overlay the logo with a button or link
   */
  overlayLogoWith?: React.ReactNode;
  /**
   * flag to display CTA
   */
  withCTA?: boolean;
  /**
   * CTA component
   */
  cta?: React.ReactNode;
  /**
   * Displayed scrolled styling
   */
  collapsed?: boolean;
}

export interface HamburgerContainerProps extends React.HTMLAttributes<HTMLSpanElement> {
  open: boolean;
}

export interface PageNavigationProps extends AppThemeProps {
  overlap?: boolean;
  collapsed?: boolean;
}

export interface NavbarLinksListLinkProps {
  item: NavigationItem;
  index: number;
  props: NavbarLinkProps;
}

const PageNavigation = styled.header<PageNavigationProps>`
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  z-index: 1;

  ${minMedia.desktop`
    ${css`
      z-index: 1;
      background-color: ${colors.white};
      max-height: ${navbarOpenHeight}px;

      transition: max-height 0.3s ease;

      ${({ collapsed }: PageNavigationProps) =>
        collapsed &&
        css`
          max-height: ${navbarClosedHeight}px;
        `}

      ${({ overlap }: PageNavigationProps) =>
        overlap &&
        css`
          max-height: ${navbarClosedHeight}px;
        `}
    `}
  `}

  .headroom {
    z-index: 1;
    background-color: ${({ theme }) => theme.navbar.mobile.bgColor};
    transition: transform 200ms ease-in-out;

    ${minMedia.desktop`
        ${css`
          background-color: ${colors.white};
        `}
      `}
  }
  .headroom--unfixed {
    transform: translateY(0);
  }

  .headroom--unpinned {
    transform: translateY(-100%);
  }
  .headroom--pinned {
    transform: translateY(0%);
  }
`;

const Spacer = styled.div<PageNavigationProps>`
  height: ${mobileNavbarHeight}px;

  ${minMedia.desktop`
    ${css`
      height: ${({ collapsed }: PageNavigationProps) => (collapsed ? navbarClosedHeight : navbarOpenHeight)}px;
    `}
  `}
`;

const LayoutInner = styled.nav<PageNavigationProps>`
  display: flex;
  align-items: center;
  justify-content: ${({ theme }: PageNavigationProps) => theme.navbar.layoutInner?.justifyContent ?? 'space-between'};
  height: 100%;
  width: 100%;
  box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px;

  ${minMedia.desktop`
    ${css`
      box-shadow: none;

      ${({ overlap }: PageNavigationProps) =>
        overlap &&
        css`
          box-shadow: rgba(0, 0, 0, 0.2) 0 1px 2px;
        `}
    `}
  `}
`;

// TODO: We have to manually type the theme, I suspect this is because we're using outdated typings
// The generic passed does not actually get passed and the generic theme from styled-components is used instead
// This should be fixed in the future
export const LogoContainer = styled.div<PageNavigationProps>`
  position: relative;
  min-height: ${({ theme }) => theme.navbar.mobile.minHeight}px;
  ${minMedia.desktop`
    ${css`
      display: flex;
      align-items: center;
      width: 490px;
      transition: 0.3s min-height ease;
      min-height: ${({ overlap }: PageNavigationProps) => (overlap ? navbarClosedHeight : navbarOpenHeight)}px;
      padding-left: ${({ theme }: PageNavigationProps) =>
        theme.navbar.logoContainer?.desktopMinMedia.paddingLeft ?? spacing[10]};
      height: ${({ theme }: PageNavigationProps) => theme.navbar.logoContainer?.desktopMinMedia.height};
      width: ${({ theme }: PageNavigationProps) => theme.navbar.logoContainer?.desktopMinMedia.width};
      justify-content: ${({ theme }: PageNavigationProps) =>
        theme.navbar.logoContainer?.desktopMinMedia.justifyContent};
    `}
  `}

  ${maxMedia.desktop`
    ${css`
      display: ${({ theme }: PageNavigationProps) => theme.navbar.logoContainer?.desktopMaxMedia.display};
      align-items: ${({ theme }: PageNavigationProps) => theme.navbar.logoContainer?.desktopMaxMedia.alignItems};
    `}
  `}

  }

  & > a,
  & > button,
  & > div {
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
  }
`;

const NavbarLinksListContainer = styled.ul`
  ${minMedia.desktop`
    ${css`
      margin-right: ${spacing[10]};
      padding-left: 0;
    `}
  `}
`;

const IconContainer = styled(Button).attrs({ 'aria-label': 'Navigation' })`
  background: transparent;
  display: ${({ theme }) => theme.navbar.iconContainer.display};
  height: ${mobileNavbarHeight}px;
  width: ${mobileNavbarHeight}px;
  font-size: 24px;
  justify-content: center;
  align-items: center;
  border-radius: 0;
`;

const LargeDeviceNavbar = styled.div`
  display: none;

  ${minMedia.desktop`
    ${`
      display:block;
    `}
  `}
`;

const SmallDeviceNavbar = styled.div`
  display: block;

  ${minMedia.desktop`
    ${`
      display:none;
    `}
  `}
`;

const NavbarLiteWrapper = ({ collapsed = false }: React.PropsWithChildren<NavbarProps>) => {
  const overThreshold = useScrollThreshold(20);
  const theme = useThemeContext();

  return (
    <>
      <PageNavigation overlap={overThreshold} collapsed={collapsed} theme={theme}>
        <ConditionalWrapper
          condition={typeof theme.navbar.href !== 'undefined'}
          wrapper={(children) => <a href={theme?.navbar?.href}>{children}</a>}
        >
          <Headroom
            wrapperStyle={{ maxHeight: overThreshold ? `${navbarClosedHeight}px` : `${navbarOpenHeight}px` }}
            disableInlineStyles
          >
            <LargeDeviceNavbar>
              <LayoutInner data-automation="ZA.navbar-desktop" overlap={overThreshold} theme={theme}></LayoutInner>
            </LargeDeviceNavbar>
            <SmallDeviceNavbar>
              <LayoutInner data-automation="ZA.navbar-mobile" overlap={overThreshold} theme={theme}></LayoutInner>
            </SmallDeviceNavbar>
          </Headroom>
        </ConditionalWrapper>
      </PageNavigation>
      <Spacer collapsed={collapsed} />
    </>
  );
};

export default NavbarLiteWrapper;
