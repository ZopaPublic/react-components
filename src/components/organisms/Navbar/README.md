### Summary

`<Navbar>` renders the base navigation bar layout.

It's meant to be used with [`<Navbar.LinksList />`](#/Components/Organisms/Navbar/NavbarLinksList). It uses [react-headroom](https://github.com/KyleAMathews/react-headroom) to show/hide the navbar depending on the scroll direction
