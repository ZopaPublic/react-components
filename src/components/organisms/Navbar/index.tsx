import NavbarWrapper from './Navbar/Navbar';
import NavbarAction from './NavbarAction/NavbarAction';

export default Object.assign(NavbarWrapper, { Action: NavbarAction });

export { navbarLinkStyles } from './NavbarLink/NavbarLink';
