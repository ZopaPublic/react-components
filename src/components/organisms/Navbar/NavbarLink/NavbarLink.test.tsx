import React from 'react';
import { render } from '@testing-library/react';
import Navbar from '..';

describe('<Navbar.Link />', () => {
  it('should render component with default props', () => {
    const { container } = render(<Navbar.Link />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render component with specific props', () => {
    const { container } = render(<Navbar.Link active open withChevron isDropdownLink />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
