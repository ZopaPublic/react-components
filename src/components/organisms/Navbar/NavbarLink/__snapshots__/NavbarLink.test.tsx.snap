// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navbar.Link /> should render component with default props 1`] = `
.c0 {
  background-color: transparent;
  font-size: inherit;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  line-height: inherit;
  color: #3B46C4;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  border: none;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  color: auto;
}

.c0:hover,
.c0:active {
  color: #21247F;
}

.c0:hover svg path,
.c0:active svg path {
  fill: #21247F;
}

.c0:hover {
  color: auto;
}

.c0:hover svg path {
  fill: auto;
}

.c0:active {
  color: auto;
}

.c0:active svg path {
  fill: auto;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  color: #3B46C4;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 12px 16px 16px;
}

.c1:focus {
  outline: none;
  box-shadow: 0px 0px 4px 0px #3B46C4;
  border-radius: 4px;
}

.c1:active,
.c1:hover {
  color: #21247F;
  opacity: 0.88;
}

@media (max-width:1279px) {
  .c1 {
    width: 100%;
  }
}

<a
  class="c0 c1"
  color="#3B46C4"
  data-automation="ZA.navbar-item"
/>
`;

exports[`<Navbar.Link /> should render component with specific props 1`] = `
.c0 {
  background-color: transparent;
  font-size: inherit;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  line-height: inherit;
  color: #3B46C4;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  -webkit-user-select: auto;
  -moz-user-select: auto;
  -ms-user-select: auto;
  user-select: auto;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  padding: 0;
  border: none;
  font-family: "Open Sans",Roboto,Helvetica,Arial,sans-serif;
  font-weight: 600;
  color: auto;
}

.c0:hover,
.c0:active {
  color: #21247F;
}

.c0:hover svg path,
.c0:active svg path {
  fill: #21247F;
}

.c0:hover {
  color: auto;
}

.c0:hover svg path {
  fill: auto;
}

.c0:active {
  color: auto;
}

.c0:active svg path {
  fill: auto;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 600;
  color: #3B46C4;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  padding: 12px 16px 16px;
}

.c1:focus {
  outline: none;
  box-shadow: 0px 0px 4px 0px #3B46C4;
  border-radius: 4px;
}

.c1:active,
.c1:hover {
  color: #21247F;
  opacity: 1;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
}

.c3 {
  display: none;
}

@media (max-width:1279px) {
  .c1 {
    width: 100%;
    color: #2C3236;
    font-size: 14px;
    font-weight: 700;
    text-transform: uppercase;
  }
}

@media (min-width:1280px) {
  .c1 {
    padding: 12px 8px;
    width: calc(100% - 16px);
    margin-left: 8px;
    margin-right: 8px;
    border-radius: 8px;
  }

  .c1:hover {
    background-color: #E9EAFA;
  }
}

@media (min-width:1280px) {
  .c3 {
    display: -webkit-inline-box;
    display: -webkit-inline-flex;
    display: -ms-inline-flexbox;
    display: inline-flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    font-size: 12px;
    -webkit-transition: -webkit-transform 0.3s;
    -webkit-transition: transform 0.3s;
    transition: transform 0.3s;
    -webkit-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }
}

<a
  class="c0 c1"
  color="#3B46C4"
  data-automation="ZA.navbar-item"
>
  <span
    class="c2"
  />
  <span
    class="c3"
    open=""
  >
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-chevron-down fa-w-14 "
      color="#818F9B"
      data-icon="chevron-down"
      data-prefix="fas"
      focusable="false"
      height="12px"
      role="img"
      viewBox="0 0 448 512"
      width="12px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        fill="currentColor"
      />
    </svg>
  </span>
</a>
`;
